mcda/electre/outranking.py

```diff
@@ -0,0 +1,254 @@
+        )
+    ]
+
+
+def crisp_outranking_coal_marginal(
+    concordance_comprehensive: NumericValue,
+    discordance_comprehensive: NumericValue,
+    concordance_cutting_level: NumericValue,
+    discordance_cutting_level: NumericValue,
+) -> bool:
+    """Constructs a crisp outranking relation, based on
+    comprehensive concordance C(a, b) and comprehensive
+    discordance D(a, b) indices.
+
+    :param concordance_comprehensive: comprehensive concordance of two alternatives
+    :param discordance_comprehensive: comprehensive concordance of two alternatives
+    :param concordance_cutting_level: concordance majority threshold (cutting level)
+    :param discordance_cutting_level: discordance majority threshold (cutting level)
+
+    :return: ``True`` if a outranks b, ``False`` otherwise
+    """
+    _check_indice_value_interval(concordance_comprehensive, "comprehensive concordance")
+    _check_indice_value_interval(discordance_comprehensive, "comprehensive discordance")
+    _check_indice_value_interval(
+        concordance_cutting_level, "concordance majority threshold", minimal_val=0.5
+    )
+    _check_indice_value_interval(
+        discordance_cutting_level, "discordance majority threshold", include_min=False
+    )
+    return (
+        concordance_comprehensive >= concordance_cutting_level
+        and discordance_comprehensive < discordance_cutting_level
+    )
+
+
+def crisp_outranking_coal(
+    concordance_comprehensive_table: List[List[NumericValue]],
+    discordance_comprehensive_table: List[List[NumericValue]],
+    concordance_cutting_level: NumericValue,
+    discordance_cutting_level: NumericValue,
+) -> List[List[bool]]:
+    """Constructs a crisp outranking relations, based on
+    comprehensive concordance C(a, b) and comprehensive
+    discordance D(a, b) indices.
+
+    :param concordance_comprehensive_table: comprehensive concordance table
+    :param discordance_comprehensive_table: comprehensive discordance table
+    :param concordance_cutting_level: concordance majority threshold (cutting level)
+    :param discordance_cutting_level: discordance majority threshold (cutting level)
+
+    :return: Boolean table the same size as the concordance and discordance tables
+    """
+    return [
+        [
+            crisp_outranking_coal_marginal(
+                concordance_comprehensive,
+                discordance_comprehensive,
+                concordance_cutting_level,
+                discordance_cutting_level,
+            )
+            for concordance_comprehensive, discordance_comprehensive in zip(
+                concordance_table_row, discordance_table_row
+            )
+        ]
+        for concordance_table_row, discordance_table_row in zip(
+            concordance_comprehensive_table, discordance_comprehensive_table
+        )
+    ]
+
+
+def outranking_relation_marginal(
+    crisp_outranking_1: bool,
+    crisp_outranking_2: bool,
```

Comment:
I would prefer
```suggestion
    crisp_outranking_ab: bool,
    crisp_outranking_ba: bool,
```
respectively